ui/components/app/gas-timing/gas-timing.component.js

```diff
@@ -1,12 +1,66 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
 import classNames from 'classnames';
 
+import { useGasFeeEstimates } from '../../../hooks/useGasFeeEstimates';
+import { useGasFeeInputs } from '../../../hooks/useGasFeeInputs';
+import { I18nContext } from '../../../contexts/i18n';
+
 import Typography from '../../ui/typography/typography';
 import { TYPOGRAPHY } from '../../../helpers/constants/design-system';
-import InfoTooltip from '../../ui/info-tooltip';
 
-export default function GasTiming({ text, tooltipText, attitude }) {
+// Once we reach this second threshold, we switch to minutes as a unit
+const SECOND_CUTOFF = 90;
+
+export default function GasTiming() {
+  const { gasFeeEstimates, isGasEstimatesLoading } = useGasFeeEstimates();
+  const { maxPriorityFeePerGas } = useGasFeeInputs();
+
+  const t = useContext(I18nContext);
+
+  // Shows "seconds" as unit of time if under SECOND_CUTOFF, otherwise "minutes"
+  const toHumanReadableTime = (milliseconds = 1) => {
+    const seconds = Math.ceil(milliseconds / 1000);
+    if (seconds <= SECOND_CUTOFF) {
+      return t('gasTimingSeconds', [seconds]);
+    }
+    return t('gasTimingMinutes', [Math.ceil(seconds / 60)]);
+  };
+
+  // Don't show anything if we don't have enough information
+  if (isGasEstimatesLoading || maxPriorityFeePerGas < 1) {
```

Comment:
What if `maxPriorityFeePerGas` is 0.5? Or any other number greater than 0 but less than 1? Then we should still show something right? Or is 1 a protocol enforced minimum to maxPriorityFeePerGas?

Reply:
Good point, removed this check!